script.js

```diff
@@ -1,4 +1,242 @@
+var advancingPlayers = []; // tracks the players who advance to next round in 'knockout' version
+var prevNum;
+var prevPlayerNum;
+var newMatch = true; // true if player is 2nd player of duel, false if 1st player
+var newGame = true; // true if new game, before any round has been played
+
+// Function for random match up
+// Returns randomly sorted player numbers, the first pair play against each other
+// the next pair and so on
+var randomMatchup = function (arrPlayers) {
+  var arrMatchups = arrPlayers.slice();
+  arrMatchups.sort(function (a, b) {
+    return 0.5 - Math.random();
+  });
+  return arrMatchups;
+};
+
+// Function for dice roll
+var randomDice = function () {
+  return Math.ceil(Math.random() * 6);
+};
+
+// Function for auto-generate dice number
+var autoGenerate = function (numOfRolls, gameMode) {
+  diceRolls = [];
+  for (var counter = 0; counter < numOfRolls; counter += 1) {
+    diceRolls.push(randomDice());
+  }
+
+  // sort the dice numbers in ascending order (default)
+  var diceRollsSorted = diceRolls.slice().sort();
+
+  // if game mode is 'highest' then sort in descending order
+  if (gameMode == "highest") {
+    diceRollsSorted = diceRollsSorted.reverse();
+  }
+
+  // auto-generate the highest / lowest number
+  // 'normal' version: compute the running score for the current player
+  // 'knockout' version: += 1 if player wins the round
+  var autoNumber = Number(diceRollsSorted.join(""));
+  return autoNumber;
+};
+
+// function to return sorted leaderboard given unsorted leaderboard and gameMode
+var leaderboardRanked = function (leaderboard, gameMode) {
+  var leaderboardSorted = leaderboard.slice();
+
+  if (gameMode == "highest") {
+    leaderboardSorted.sort(function (a, b) {
+      return b[1] - a[1];
+    });
+  } else if (gameMode == "lowest") {
+    leaderboardSorted.sort(function (a, b) {
+      return a[1] - b[1];
+    });
+  }
+  return leaderboardSorted;
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  if (input == "highest" || input == "lowest") {
+    gameMode = input;
+    return `Game mode has been set to ${gameMode} combined number.`;
+  }
+
+  if (input == "normal" || input == "knockout") {
+    gameVer = input;
+    gameState = 0;
+    return `Game has switched to ${gameVer} version. Enter number of players to continue.`;
+  }
+
+  switch (gameState) {
+    case 0:
+      // input validation
+      if (!Number(input)) {
+        return "Enter a number to continue";
+      }
+
+      numOfPlayers = input;
+      gameState = 1;
+      leaderboard = [];
+      for (var user = 1; user <= numOfPlayers; user += 1) {
+        leaderboard.push([user, 0]);
+      }
+      var msg = `There are ${numOfPlayers} players for this game. `;
+      return msg;
+
+    case 1:
+      // input validation
+      if (!Number(input)) {
+        return "Enter a number to continue";
+      }
+      numOfRolls = input;
+      gameState = 2;
```

Comment:
just nitpicking, but try using words instead of numbers so that your code is more readable, especially for something like gameState